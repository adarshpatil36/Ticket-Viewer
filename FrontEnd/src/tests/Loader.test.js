import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";

import Loader from "../components/Loader";

describe("LoaderComponent testing", () => {
  it("renders LoaderComponent component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Loader />, div);
  });
  it("renders LoaderComponent with proper label", () => {
    const { queryByLabelText } = render(<Loader />);
    expect(queryByLabelText).toBeTruthy();
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Loader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
