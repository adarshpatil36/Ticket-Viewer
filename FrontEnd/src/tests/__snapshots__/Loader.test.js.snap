// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderComponent testing renders correctly 1`] = `
Array [
  <div
    className="spinner-border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Loading...
    </span>
  </div>,
  "Loading...",
]
`;
